### PR TITLE
[v6.2] docs: proxy_protocol

### PR DESCRIPTION
**Pull request by inertial-frame**
Addresses: https://github.com/gravitational/teleport/issues/6508

### CODE REVIEW

**Review by klizhentas — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

File: docs/pages/config-reference.mdx [hunk 1]
```diff
@@ -156,6 +156,12 @@ auth_service:
     # certificates and keys (may need to wipe out /var/lib/teleport directory)
     cluster_name: "main"
 
+    # ProxyProtocol enables support for HAProxy proxy protocol when it is turned 'on'.
```
Comment (webvictim) [inline docs/pages/config-reference.mdx:159]:
```suggestion
    # ProxyProtocol enables support for HAProxy proxy protocol version 1 when it is turned 'on'.
```

Teleport doesn't support v2 (the binary version), only v1 (the plain text version). Would be useful for people to know.

File: docs/pages/config-reference.mdx [hunk 2]
```diff
@@ -156,6 +156,12 @@ auth_service:
...
+    # We recommend it be enabled by an administrator since knowing
+    # whether the service is in front of trusted load balancer is relevant.
```
Comment (webvictim) [inline docs/pages/config-reference.mdx:161]: This sentence doesn't make a lot of sense if the default value is `on` IMO as it seems like it's always going to be enabled?

Comment (webvictim) [inline docs/pages/config-reference.mdx:161]:
```suggestion
    # whether the service is in front of a trusted load balancer is relevant.
```

Comment (inertial-frame) [reply docs/pages/config-reference.mdx]: Yeah, after a second reading I agree. I think the intent of the original comment here is just to verify with someone knowledgeable. I've tweaked this just a bit:

`Verify whether the service is in front of a trusted load balancer.`

**Review by klizhentas — APPROVED** [review]

**Review by benarent — APPROVED** [review]

**Pull request merged**